disko

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+set -euo pipefail
+set -x
+
+readonly libexec_dir="${0%/*}"
+
+# mount was chosen as the default mode because it's less destructive
+mode=mount
+nix_args=()
+
+abort() {
+  echo "aborted: $*" >&2
+  exit 1
+}
+
+## Main ##
+
+while [[ $# -gt 1 ]]; do
+  case "$1" in
+    -m | --mode)
+      mode=$2
+      shift
+      ;;
+    --argstr | --arg)
+      nix_args+=("$1" "$2" "$3")
+      shift
+      shift
+      ;;
+    *)
+      if [ -z ${disko_config+x} ]; then
+        disko_config=$1
+      else
+        abort "disko config is already set"
```

Comment:
I would also add a `usage()` function that prints all accepted parameters in case and print it before showing the error.
Also when a user specifies `--help`.

Reply:
yeah will do. will also have to add tests for the module and the new cli